examples/advanced/llm_hf/README.md

```diff
@@ -0,0 +1,131 @@
+# LLM Tuning via HuggingFace SFT Trainer
+This example shows how to use [NVIDIA FLARE](https://nvidia.github.io/NVFlare) for Large Language Models (LLMs) tuning tasks. It illustrates how to adapt a local training script with [HuggingFace](https://huggingface.co/) trainer to NVFlare.
+
+## Introduction 
+This example illustrates both supervised fine-tuning (SFT) and parameter-efficient fine-tuning (PEFT) using the [SFT Trainer](https://huggingface.co/docs/trl/sft_trainer) from [HuggingFace](https://huggingface.co/).
+We used the [Llama-2-7b model](https://huggingface.co/meta-llama/Llama-2-7b) to showcase the functionality of federated SFT and PEFT, allowing HuggingFace models to be trained and adapted with NVFlare. 
+Mainly on two fronts:
+- Adapt local HF training scripts to federated application
+- Handling large model weights (~26 GB for Llama-2-7b model), this is supported by NVFlare infrastructure, and does not need any code change.
+
+We perform the following exeperiments on two 80GB A100 GPUs, PEFT only needs 1 GPU, while SFT needs both GPUs. Less computation resources will be needed if smaller models are used, simply replace Llama-2-7b-hf with other options from HuggingFace.
+
+## Setup
+Please make sure you set up virtual environment following [example root readme](../../README.md).
+Install additional requirements:
+```
+python3 -m pip install -r requirements.txt
+```
+
+## Model and Data Preparation
+We first download the model and save it to the `model` folder, note that approved access to the model is required
+```
+mkdir model
+cd model
+git clone https://huggingface.co/meta-llama/Llama-2-7b-hf
+cd ..
+```
+
+We then download and preprocess (to be consistent with our [NeMo example](../../../integration/nemo/examples/supervised_fine_tuning), we follow the same preprocessing steps) the dataset [databricks-dolly-15k](https://huggingface.co/datasets/databricks/databricks-dolly-15k) for this example. 
+```
+mkdir dataset
+cd dataset
+git clone https://huggingface.co/datasets/databricks/databricks-dolly-15k
+cd ..
+mkdir dataset/dolly
+python ./utils/preprocess_dolly.py --training_file dataset/databricks-dolly-15k/databricks-dolly-15k.jsonl --output_dir dataset/dolly
+```
+
+## Centralized Training
+### One-call training
+Centralized trainings, as the baseline for comparison with FL results, are done with the following command:
+```
+python3 ./utils/hf_sft_peft.py --output_path ./workspace_centralized/llama2-7b-dolly-sft --mode 0
+python3 ./utils/hf_sft_peft.py --output_path ./workspace_centralized/llama2-7b-dolly-peft --mode 1
+```
+### Adaptation Step 1: iterative training
+To adapt the centralized training script to federated application, we first need to "break" the single call to `trainer.train()` into iterative calls, one for each round of training.
```

Comment:
Note that, this "break" into iterative script is NOT necessary.
The iterative/loop pattern is only needed when we use "launch_once" = True.

Reply:
Added "under "launch_once" setting"

Reply:
@ZiyueXu77 I am using this example to guide users: https://github.com/NVIDIA/NVFlare/tree/main/examples/hello-world/ml-to-fl/np

@SYangster @nvkevlu should we add something to explain this "launch_once"?